cpp/ql/src/experimental/Likely Bugs/DerefNullResult.ql

```diff
@@ -0,0 +1,34 @@
+/**
+ * @name Null dereference from a function result
+ * @description A function parameter is dereference,
+ *              while it comes from a function that may return NULL,
+ *              and is not checked for nullness by the caller.
+ * @kind problem
+ * @id cpp/deref-null-result
+ * @problem.severity recommendation
+ * @tags reliability
+ *       security
+ *       external/cwe/cwe-476
+ */
+
+import cpp
+import semmle.code.cpp.dataflow.new.DataFlow
+
+from Function nuller, Parameter pd, FunctionCall fc, Variable v
+where
+  mayReturnNull(nuller) and
+  functionDereferences(pd.getFunction(), pd.getIndex()) and
+  // there is a function call which will deref parameter pd
+  fc.getTarget() = pd.getFunction() and
+  // the parameter pd comes from a variable v
+  DataFlow::localFlow(DataFlow::exprNode(v.getAnAccess()),
+    DataFlow::exprNode(fc.getArgument(pd.getIndex()))) and
+  // this variable v was assigned by a call to the nuller function
+  v.getAnAssignedValue() = nuller.getACallToThisFunction() and
+  // this variable v is not accessed for an operation (check for NULLness)
+  not exists(VariableAccess vc |
+    vc.getTarget() = v and
+    (vc.getParent() instanceof Operation or vc.getParent() instanceof IfStmt)
+  )
```

Comment:
You could do wrap `v.getAnAssignedValue()` in a `unique` such as:
```
unique(| | v.getAnAssignedValue()) = nuller.getACallToThisFunction()
```
Note, however, that it may be better to simply remove the need for `v` completely as I suggested [here](https://github.com/github/codeql/pull/16524/files/8ace9da14a9ee331b841e2dfea9253e30dcb71de#r1607977870) 🙂

Reply:
Just did that as the other solution does not work out so good ;-)